lib/StringClient.js

```diff
@@ -12,12 +12,15 @@ var util = require('util');
 
 var HttpClient = require('./HttpClient');
 
+var nodeMajorVer = parseInt(process.versions.node.split('.'), 10);
 
 // --- API
 
 function StringClient(options) {
     assert.object(options, 'options');
     assert.optionalObject(options.gzip, 'options.gzip');
+    assert.optionalBool(options.allowAlternativeContentMd5,
```

Comment:
Instead of a binary -- can we have this accept the encoding instead? This would make the API more explicit. Actually, on second thought -- what's the purpose of this? Are there reasons you'd want a different encoding from the client? If so, shouldn't the encoding be configurable by the user?

Reply:
Please see issue #173 - the problem with explicitly passing in an encoding is that it means you must know what version of node the **server** is using - i.e. it's something the client doesn't know in advance.

Reply:
FYI, I updated the issue to make this clearer:

When a Content-MD5 header is returned in the response to the restify client, the client validates the content (a md5 hash) to ensure the content is correct - if the hash does not match the Content-MD5 header then a BadDigest error is returned in the callback.

When a Node v6 client talks to a Node v4 server (and vice versa), a BadDigest error is returned even though the content is correct - as the default md5 hash generated between these two node versions are different.